packages/itwinui-react/src/core/Carousel/Carousel.tsx

```diff
@@ -122,10 +127,10 @@ export const Carousel = Object.assign(
       }
     };
 
-    const userOnSlideChange = React.useRef(onSlideChange);
+    const userOnSlideChange = useLatestRef(onSlideChange);
     React.useEffect(() => {
       userOnSlideChange.current?.(currentIndex);
-    }, [currentIndex]);
+    }, [userOnSlideChange, currentIndex]);
```

Comment:
Just wondering, why is a ref needed here? Like would something like this work?

```tsx
React.useEffect(() => {
  onSlideChange(currentIndex);
}, [currentIndex]);
```

Or am I missing something? (I tried to test it myself but having some repo setup issues. So thought I'd ask here.)

Reply:
the latest ref pattern is needed to avoid stale callbacks (the exact bug we are trying to solve). you can read more here: https://overreacted.io/making-setinterval-declarative-with-react-hooks/

what repo issues are you having?

Reply:
Thanks, I am going through that link. I am getting this error that is not allowing me to open storybook:

![image](https://github.com/iTwin/iTwinUI/assets/45748283/404e0bf4-86f7-4f34-9fd8-d34a0203e84b)

Reply:
ah, try deleting the `styles.js` folder manually before starting dev server

Reply:
Unfortunately, that didn't work. It is still giving the same error even after deleting the `packages/itwinui-react/src/styles.js` folder.

Am I missing something? Do you recommend recloning the repo?